sycl/include/sycl/device.hpp

```diff
@@ -214,8 +217,29 @@ class __SYCL_EXPORT device : public detail::OwnerLessBase<device> {
   /// type associated with the param parameter.
   ///
   /// \return device info of type described in Table 4.20.
+#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
+  template <typename Param>
+  typename detail::is_device_info_desc<Param>::return_type get_info() const {
+    // For C++11_ABI compatibility, we handle these string Param types
+    // separately.
+    if constexpr (std::is_same_v<Param, info::device::name> ||
+                  std::is_same_v<Param, info::device::vendor> ||
+                  std::is_same_v<Param, info::device::driver_version> ||
+                  std::is_same_v<Param, info::device::version> ||
+                  std::is_same_v<Param, info::device::profile>) {
+
+      detail::string_view PropertyName = typeid(Param).name();
+      detail::string Info; // Libsycl fills this param with property information
+      get_device_info(PropertyName, Info);
+      std::string DeviceInfo = Info.c_str();
+      return DeviceInfo;
+    }
+    return get_info_internal<Param>();
```

Comment:
Again, this is the dual ABI issue. We cannot call a template function in libsycl from a SYCL header file if the template parameter is a std::string.
We need to convert the template parameter to a detail::string and reconstruct it in the libsycl.

Reply:
Yes, I get it. I'm just proposing having:
```c++
template <typename Param> detail::string get_device_info();
```
instead of
```c++
void get_device_info(detail::string_view Type);
```
We would save the `typeid(Param).name()` in the call this way and the string comparison in `get_device_info` definition.